agent/local/state.go

```diff
@@ -1538,16 +1539,13 @@ func (l *State) notifyIfAliased(serviceID structs.ServiceID) {
 // critical purposes, such as logging. Therefore we interpret all errors as empty-string
 // so we can safely log it without handling non-critical errors at the usage site.
 func (l *State) aclAccessorID(secretID string) string {
-	ident, err := l.Delegate.ResolveTokenToIdentity(secretID)
```

Comment:
Isn't this doing significantly more work now? Maybe not for this PR, but could we avoid building the authorizer here? This call is fairly expensive from what I've seen in some CPU profiles

Reply:
Yes, good point! 

We only call this method when an RPC call returns either:
* `acl.ErrNotFound`
* `acl.ErrPermissionDenied` 

So it should be pretty rare, but we do have some options to make it less expensive.

I was going to suggest we could cache the AccessorID, but I noticed `ACLResolver` already caches the `Authorizer`. It seems like we cache up to 256 Authorizers (1024 on servers), so I wonder how likely it would be that we miss the cache.

We could restore the method to retrieve only the `Identity` from a token, but I'm not sure if these rare cases are enough to warrant that extra code.

We could also decide to add a new cache of `secretID` -> `AccessorID` that is local to only the `agent/local.State`.

I wonder if we could actually remove this call entirely soon. We are planning on adding the AccessorID to "permission denied" error. Presumably if the error is "acl not found", then there is no accessor ID to return (or if it's a policy that is not found, we could include the accessor ID in that error as well).

Given that we are planning on improve those errors in the next release, I wonder if we could leave this as-is, and remove this `aclAccessorID` method once we've enriched the errors on the server side. Does that sound like a good approach? I think that gives us better errors without having to do any kind of lookup for AccessorID.

Reply:
That seems pretty reasonable; can we capture this work somewhere to make sure it isn't missed?

Reply:
Good call! I opened #12233 to track this cleanup.